src/filter_tree/filter_tree.c

```diff
@@ -157,7 +157,6 @@ void _FilterTree_SubTrees
 					// break AND down to its components
 					_FilterTree_SubTrees(root->cond.left, sub_trees);
 					_FilterTree_SubTrees(root->cond.right, sub_trees);
-					rm_free((FT_FilterNode *)root);
```

Comment:
Right, if we're operating on the "original" input tree we shouldn't free this AND operation node